File: src/main/java/br/com/barbearia/api/controller/AgendamentoController.java
package br.com.barbearia.api.controller;

import br.com.barbearia.api.dto.AgendamentoRequestDTO;
import br.com.barbearia.api.dto.AgendamentoResponseDTO;
import br.com.barbearia.api.service.AgendamentoService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDate;
import java.util.List;

@CrossOrigin(origins = "http://localhost:8080")
@RestController
@RequestMapping("/agendamentos")
public class AgendamentoController {

    private final AgendamentoService service;

    public AgendamentoController(AgendamentoService service) {
        this.service = service;
    }

    @PostMapping
    public ResponseEntity<AgendamentoResponseDTO> criar(
            @RequestBody AgendamentoRequestDTO dto) {

        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(service.criar(dto));
    }

    @GetMapping
    public ResponseEntity<List<AgendamentoResponseDTO>> listar() {
        return ResponseEntity.ok(service.listar());
    }

    @GetMapping("/data/{data}")
    public ResponseEntity<List<AgendamentoResponseDTO>> listarPorData(
            @PathVariable LocalDate data) {

        return ResponseEntity.ok(service.listarPorData(data));
    }

    @PatchMapping("/{id}/confirmar")
    public ResponseEntity<Void> confirmar(@PathVariable Long id) {
        service.confirmar(id);
        return ResponseEntity.noContent().build();
    }

    @PutMapping("/{id}")
    public ResponseEntity<AgendamentoResponseDTO> atualizar(
            @PathVariable Long id,
            @RequestBody AgendamentoRequestDTO dto) {

        return ResponseEntity.ok(service.atualizar(id, dto));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> cancelar(@PathVariable Long id) {
        service.cancelar(id);
        return ResponseEntity.noContent().build();
    }
}

File: controller/agendamentoController.js
import agendamentoService from "../services/agendamentoService.js";

class AgendamentoController {

  async criar(req, res) {
    try {
      console.log("BODY:", req.body);

      const agendamento = await agendamentoService.criar(req.body);
      return res.status(201).json(agendamento);
    } catch (error) {
      console.log("ERRO:", error.message);
      return res.status(400).json({ message: error.message });
    }
  }

  async listar(req, res) {
    try {
      const agendamentos = await agendamentoService.listar();
      return res.json(agendamentos);
    } catch (err) {
      return res.status(500).json({ message: err.messsage })
    }

  }

  async listarPorData(req, res) {
    const { data } = req.params;
    const agendamentos = await agendamentoService.listarPorData(data);
    return res.json(agendamentos);
  }

  async confirmar(req, res) {
    await agendamentoService.confirmar(req.params.id);
    return res.status(204).send();
  }

  async cancelar(req, res) {
    await agendamentoService.cancelar(req.params.id);
    return res.status(204).send();
  }

  async atualizar(req, res) {
    try {
      const agendamento = await agendamentoService.atualizar(
        req.params.id,
        req.body
      );
      return res.json(agendamento);
    } catch (error) {
      return res.status(400).json({ message: error.message });
    }
  }
}

export default new AgendamentoController();